api/v1alpha1/envoygateway_types.go

```diff
@@ -571,6 +577,43 @@ type ExtensionTLS struct {
 	CertificateRef gwapiv1.SecretObjectReference `json:"certificateRef"`
 }
 
+// GRPCStatus defines grpc status codes as defined in https://github.com/grpc/grpc/blob/master/doc/statuscodes.md.
+// +kubebuilder:validation:Enum=CANCELLED;UNKNOWN;INVALID_ARGUMENT;DEADLINE_EXCEEDED;NOT_FOUND;ALREADY_EXISTS;PERMISSION_DENIED;RESOURCE_EXHAUSTED;FAILED_PRECONDITION;ABORTED;OUT_OF_RANGE;UNIMPLEMENTED;INTERNAL;UNAVAILABLE;DATA_LOSS;UNAUTHENTICATED
+type RetryableGRPCStatusCode string
+
+// RetryPolicy defines the retry policy for to use when errors are encountered in communication with the extension service.
+type RetryPolicy struct {
+	// MaxAttempts defines the maximum number of retry attempts.
+	// Default: 4
+	//
+	// +optional
+	MaxAttempts *int `json:"maxAttempts,omitempty"`
+
+	// InitialBackoff defines the initial backoff in seconds for retries, details: https://github.com/grpc/proposal/blob/master/A6-client-retries.md#integration-with-service-config.
+	// Default: 0.1s
+	//
+	// +optional
+	InitialBackoff *gwapiv1.Fraction `json:"initialBackoff,omitempty"`
```

Comment:
shouldnt the backoffs be duration ?
https://github.com/grpc/grpc-go/blob/515f377af267b59ef67eb0da965ec8eeb8be035e/internal/serviceconfig/serviceconfig.go#L157